core/transcoder/codecs.go

```diff
@@ -16,7 +16,7 @@ type Codec interface {
 	DisplayName() string
 	GlobalFlags() string
 	PixelFormat() string
-	ExtraArguments() string
+	ExtraArgumentsForLevel(l int) string
```

Comment:
Would it make sense to add your extra flags into `VariantFlags` instead, since you're doing a variant-specific configuration? Plus it already has the specific details as `HLSVariant` data.  `ExtraArguments` is really extra arguments for the codec as a whole, not for a variant.